components/brave_rewards/resources/extension/brave_rewards/background/api/tabs_api.ts

```diff
@@ -2,8 +2,45 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+const isTwitterURL = (url: URL) => {
+  return url && url.hostname.endsWith('.twitter.com')
+}
+
+const getTwitterScreenName = (url: URL) => {
```

Comment:
does this need to exclude URLs which look like screennames but actually aren't, such as `https://twitter.com/notifications`? or is that handled later?

Reply:
Yes, we exclude non-screenname paths in the C++ backend here: https://github.com/brave/brave-core/blob/master/vendor/bat-native-ledger/src/bat/ledger/internal/media/twitter.cc#L87